lib/puppet-strings/hiera/data.rb

```diff
@@ -64,13 +78,40 @@ def to_s
 
     private
 
+    def value_to_s(value)
+      if value.is_a?(Array) && value.empty?
+        return '[]'
+      elsif value.is_a?(Hash) && value.empty?
+        return '{}'
+      elsif value.nil?
+        return 'undef'
+      else
+        return value.to_s
+      end
```

Comment:
Have you considered using `inspect`?
```suggestion
      value.nil? ? 'undef' : value.inspect
```

Reply:
updated and agree on this being the wrong place